src/compiler/checker.ts

```diff
@@ -161,6 +161,8 @@ namespace ts {
 
         let jsxElementClassType: Type;
 
+        let deferredNodes: Node[];

```

Comment:
The `deferredNodes` array is used only when doing a full type check. When called from the language service (i.e. when "helicoptering in") we don't use it.